vllm/worker/worker.py

```diff
@@ -82,11 +81,12 @@ def __init__(
                 not in ["medusa", "mlp_speculator"]) \
                     else {"return_hidden_states": True}
 
-        ModelRunnerClass: Type[GPUModelRunnerBase] = ModelRunner
+        #ModelRunnerClass: Type[GPUModelRunnerBase] = ModelRunner
         if model_runner_cls is not None:
             ModelRunnerClass = model_runner_cls
-        elif self.model_config.embedding_mode:
-            ModelRunnerClass = EmbeddingModelRunner
+        else:
+            ModelRunnerClass = ModelMode.get_model_runner_cls(
+                self.model_config.model_mode)
```

Comment:
I think we can simplify this to a one-liner

```py
ModelRunnerClass = model_runner_cls or ModelMode.get_model_runner_cls(self.model_config.model_mode)
```

Reply:
Yes, get_block_space_manager_class() is not used anymore, replaced by get_block_space_manager_impl()